src/plugins/intel_gpu/src/graph/condition.cpp

```diff
@@ -139,22 +139,15 @@ static ov::PartialShape resolve_shape(const ov::PartialShape& true_pshape, const
     return ov::PartialShape(new_dims);
 }
 
-static std::vector<layout> resolve_shape(std::vector<layout>& target_list, std::vector<layout>& other_list) {
-    std::vector<layout> resolved_layout;
-    for (size_t i = 0; i < target_list.size(); i++) {
-        auto target = target_list[i];
-        auto other = other_list[i];
-        auto target_pshape  = target.get_partial_shape();
-        auto other_pshape   = other.get_partial_shape();
-        auto target_rank    = target_pshape.rank();
-        auto other_rank     = other_pshape.rank();
-        if (target_rank.get_length() == 0 && other_rank.get_length() == 1) {
-            resolved_layout.push_back({ov::PartialShape{1}, target.data_type, target.format});
-        } else {
-            resolved_layout.push_back(target);
-        }
+layout condition_inst::resolve_layout(layout& target, layout& other) {
```

Comment:
first, the name of function is somewhat ambiguous. 
Also I think we'd better to have this functionality to layout, like, to_rank(rank) something.

Reply:
Change the function name to adjust_scalar_to_1d_layout. Thanks.